addons/knowledge/static/src/js/wysiwyg/knowledge_article_link.js

```diff
@@ -103,9 +104,12 @@ const KnowledgeArticleLinkModal = Dialog.extend({
              * @param {Function} escapeMarkup
              */
             formatResult: (result, container, query, escapeMarkup) => {
-                const { display_name } = result;
+                const { display_name, subject } = result;
                 const markup = [];
                 window.Select2.util.markMatch(display_name, query.term, markup, escapeMarkup);
+                if (subject != display_name) {
```

Comment:
Above you test "subject" and not here, any specific reason?

Reply:
I will remove it from above as well, since `subject` should always be defined

Reply:
Nvm, there was a reason! In this panel, the category elements (Private, Workspace) don't have a subject